src/NCalc/LambdaExpressionVistor.cs

```diff
@@ -159,7 +159,9 @@ public override void Visit(Function function)
             switch (function.Identifier.Name.ToLowerInvariant())
             {
                 case "if":
-                    _result = L.Expression.Condition(args[0], args[1], args[2]);
+                    var if_arg1 = args[1].Type == typeof(bool) ? args[1] : L.Expression.Convert(args[1], typeof(double));

```

Comment:
isn't this gonna break for `if(true, "hello", "world")`? `arg[1]` != bool -> convert(string, double). I think you need to check if both arg[1] and arg[2] are numeric and then cast it to the common denominator type

Reply:
Thank you sklose, yes you are right. I'll fix it as soon as i can.

Reply:
I updated the implementation and added more tests. Let me know what you think.